app/grandchallenge/evaluation/models.py

```diff
@@ -1710,6 +1712,54 @@ class EvaluationGroundTruthGroupObjectPermission(GroupObjectPermissionBase):
     )
 
 
+class EvaluationInputSetManager(models.Manager):
+    def create(
+        self,
+        *,
+        inputs,
+        **kwargs,
+    ):
+        obj = get_existing_evaluation_input_set_for_inputs(
+            inputs=inputs,
+        )
+        if not obj:
+            obj = super().create(**kwargs)
+            obj.values.set(inputs)
+
+        return obj
+
+
+class EvaluationInputSet(UUIDModel):
+    values = models.ManyToManyField(
+        ComponentInterfaceValue, related_name="eval_inputs"
```

Comment:
We don't really need to use an abbreviation, the full name is clearer.

```suggestion
        ComponentInterfaceValue, related_name="evaluation_inputs"
```

Reply:
I think this clashes with the related name for `inputs` on the `Phase` model...